### PR TITLE
fix(core): use preferred installer on migrate

### LINKED ISSUE #2695: nx migrate does not work if only yarn is installed (no npm)
## Expected Behavior

Should find one which one (npm or yarn) is used.

## Current Behavior

Does not use yarn, still tries to use npm.

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Install only yarn (without npm)
2. nx migrate...

### Context

nx report:
```
yarn run v1.22.4
$ nx report

>  NX  Report complete - copy this into the issue template

  @nrwl/angular : Not Found
  @nrwl/cli : 9.0.4
  @nrwl/cypress : 9.0.4
  @nrwl/eslint-plugin-nx : 9.0.4
  @nrwl/express : Not Found
  @nrwl/jest : 9.0.4
  @nrwl/linter : 9.0.4
  @nrwl/nest : 9.0.4
  @nrwl/next : Not Found
  @nrwl/node : 9.0.4
  @nrwl/react : 9.0.4
  @nrwl/schematics : Not Found
  @nrwl/tao : 9.0.4
  @nrwl/web : 9.0.4
  @nrwl/workspace : 9.0.4
  typescript : 3.8.3

✨  Done in 0.49s.
```

### Failure Logs

```
Fetching meta data about packages.
It may take a few minutes.
Fetching @nrwl/workspace@latest
The migrate command failed. Try the following to migrate your workspace:
> npm install --save-dev @nrwl/workspace@latest
> nx migrate @nrwl/workspace@latest --from="@nrwl/workspace@9.0.4"
This will use the newest version of the migrate functionality, which might have your issue resolved.
----------------------------------------------------------------------------------------------------
Command failed: npm install @nrwl/workspace@latest --prefix=/tmp/tmp-6280089vkLxKnRzUe6
/bin/sh: npm: command not found

/home/theaifam5/Documents/Projects/node_modules/yargs/yargs.js:1109
      else throw err
           ^

Error: Command failed: ./node_modules/.bin/tao migrate @nrwl/workspace
    at checkExecSyncError (child_process.js:611:11)
    at Object.execSync (child_process.js:647:15)
    at Object.handler (/home/theaifam5/Documents/Projects/node_modules/@nrwl/workspace/src/command-line/nx-commands.js:76:21)
    at Object.runCommand (/home/theaifam5/Documents/Projects/node_modules/yargs/lib/command.js:235:44)
    at Object.parseArgs [as _parseArgs] (/home/theaifam5/Documents/Projects/node_modules/yargs/yargs.js:1022:30)
    at Object.get [as argv] (/home/theaifam5/Documents/Projects/node_modules/yargs/yargs.js:965:21)
    at Object.initLocal (/home/theaifam5/Documents/Projects/node_modules/@nrwl/cli/lib/init-local.js:20:79)
    at Object.<anonymous> (/home/theaifam5/Documents/Projects/node_modules/@nrwl/cli/bin/nx.js:12:18)
    at Module._compile (internal/modules/cjs/loader.js:1147:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1167:10) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 628008,
  stdout: null,
  stderr: null
}
```
### Other
https://github.com/nrwl/nx/blob/0b7535ae92b2aaa72fa1df3c9ea45d96aee5f8c4/packages/tao/src/commands/migrate.ts#L405

### PR DESCRIPTION
**Pull request by bekos**
ISSUES CLOSED: #2695

### CODE REVIEW

**Comment by vsavkin** [comment]
The build fails :)

File: packages/tao/src/shared/detect-package-manager.ts [hunk 1]
```diff
@@ -21,28 +12,29 @@ function isPackageManagerInstalled(packageManager: string) {
...
-  if (hostTree.get('workspace.json')) {
-    const workspaceJson: { cli: { packageManager: string } } = JSON.parse(
-      hostTree.read('workspace.json').toString()
+export function detectPackageManager() {
+  const workspaceJsonPath = [`workspace.json`, `angular.json`].find((p) =>
+    existsSync(p)
+  );
+
+  if (workspaceJsonPath) {
+    const workspaceJson = JSON.parse(
```
Comment (FrozenPandaz) [inline packages/tao/src/shared/detect-package-manager.ts:21]: Idk why this wasn't using it before but we have a function called `readJsonFile` that also handles comments in JSONs.

File: packages/tao/src/shared/detect-package-manager.ts [hunk 2]
```diff
@@ -21,28 +12,29 @@ function isPackageManagerInstalled(packageManager: string) {
   }
 }
 
-export async function detectPackageManager(
-  host: virtualFs.Host<Stats>
```
Comment (FrozenPandaz) [inline packages/tao/src/shared/detect-package-manager.ts:25]: I don't think it was necessary to use the `Host` but is there a particular reason why we don't use it anymore?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/tao/src/shared/detect-package-manager.ts (hunk 1 above)
Comment (bekos) [reply packages/tao/src/shared/detect-package-manager.ts:21]: `readJsonFile` is part of the `@nrwl/workspace` and couldn't import it here 😃 do you want me to strip comments also?

File: packages/tao/src/shared/detect-package-manager.ts (hunk 2 above)
Comment (bekos) [reply packages/tao/src/shared/detect-package-manager.ts:25]: One reason is to try to align with `detectPackageManager` of `@nrwl/workspace` which doesn't accept `Host` and the other reason is that it wasn't available on the `migrate` 🤔

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**